src/main/java/com/merixstudio/rq2877270/mrpotato/pokemon/UserFavouritesController.java

```
@@ -0,0 +1,58 @@
+package com.merixstudio.rq2877270.mrpotato.pokemon;
+
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class UserFavouritesController {
+    private final UserFavouritesRepository repository;
+
+    UserFavouritesController(UserFavouritesRepository repository) {
+        this.repository = repository;
+    }
+
+    @GetMapping("/favourites")
+    List<UserFavourites> all() {
+        return repository.findAll();
+    }
+
+    @PostMapping("/favourites")
+    UserFavourites newFavourite(@RequestBody UserFavourites newFavourite) {
+        return repository.save(newFavourite);
+    }
+
+    @PostMapping("/favourites/gem")
+    void generateSomeFavourites() {
+        for(int i=0; i < 10; i++) {
+            System.out.println("Generating favourite pairs: user " + i + " and pokemon " + 1);
+            UserFavourites newFavourite = new UserFavourites();
+            newFavourite.setUserId((long) i);
+            newFavourite.setPokemonId((long) i);
+            repository.save(newFavourite);
+        }
+    }
+
+    // Single item
+    @GetMapping("/favourites/{id}")
+    UserFavourites one(@PathVariable Long id) {
+        return repository.findById(id)
+                .orElseThrow(() -> new UserFavouriteNotFoundException(id));
```

Exception is not handled so API will return 500 code instead 404.